### PR TITLE
ATS-669: Parameterize T-Engines transformer execution locations

### PR DESCRIPTION
**Pull request by KikoUK**

### CODE REVIEW

**Comment by montgolfiere** [comment]
For next time, use ATS-xyz as branch name (rather than /story/ATS-xyz) ? See also:

https://github.com/Alfresco/alfresco-transform-core/blob/master/.travis.yml#L24

File: alfresco-transformer-base/src/main/java/org/alfresco/transformer/config/GlobalProperties.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
+package org.alfresco.transformer.config;
```
Comment (montgolfiere) [inline alfresco-transformer-base/src/main/java/org/alfresco/transformer/config/GlobalProperties.java:1]: The T-Base should include anything specific to external transformers  since it is also used to build custom T-Engines (that are unrelated to I/M, L/O, P/R).

File: alfresco-transform-imagemagick/alfresco-transform-imagemagick/src/main/java/org/alfresco/transformer/executors/ImageMagickCommandExecutor.java [hunk 2]
```diff
@@ -35,9 +37,18 @@
  */
 public class ImageMagickCommandExecutor extends AbstractCommandExecutor
 {
-    private static final String ROOT = "/usr/lib64/ImageMagick-7.0.7";
-    private static final String DYN = ROOT + "/lib";
-    private static final String EXE = "/usr/bin/convert";
+    private static String ROOT;
+    private static String DYN;
+    private static String EXE;
```
Comment (montgolfiere) [inline alfresco-transform-imagemagick/alfresco-transform-imagemagick/src/main/java/org/alfresco/transformer/executors/ImageMagickCommandExecutor.java:42]: Could/should these be "final" (rather than "static") ?

File: alfresco-transform-pdf-renderer/alfresco-transform-pdf-renderer/src/main/java/org/alfresco/transformer/executors/PdfRendererCommandExecutor.java [hunk 3]
```diff
@@ -35,7 +35,14 @@
  */
 public class PdfRendererCommandExecutor extends AbstractCommandExecutor
 {
-    private static final String EXE = "/usr/bin/alfresco-pdf-renderer";
+    private static String EXE;
```
Comment (montgolfiere) [inline alfresco-transform-pdf-renderer/alfresco-transform-pdf-renderer/src/main/java/org/alfresco/transformer/executors/PdfRendererCommandExecutor.java:38]: Could/should this be "final" (rather than "static") ?

File: alfresco-transform-libreoffice/alfresco-transform-libreoffice/src/main/java/org/alfresco/transformer/executors/LibreOfficeJavaExecutor.java [hunk 4]
```diff
@@ -53,12 +53,13 @@
     private static final Logger logger = LoggerFactory.getLogger(LibreOfficeJavaExecutor.class);
 
     private static final int JODCONVERTER_TRANSFORMATION_ERROR_CODE = 3088;
-    private static final String OFFICE_HOME = "/opt/libreoffice6.1";
+    private static String OFFICE_HOME;
```
Comment (montgolfiere) [inline alfresco-transform-libreoffice/alfresco-transform-libreoffice/src/main/java/org/alfresco/transformer/executors/LibreOfficeJavaExecutor.java:56]: Could/should this be "final" (rather than "static") ?

File: alfresco-transformer-base/src/main/resources/application.yaml [hunk 5]
```diff
@@ -48,3 +48,12 @@ management:
 
 container:
   name: ${HOSTNAME:t-engine}
+
+transform:
```
Comment (montgolfiere) [inline alfresco-transformer-base/src/main/resources/application.yaml:52]: The T-Base should include anything specific to external transformers since it is also used to build custom T-Engines (that are unrelated to I/M, L/O, P/R).

File: alfresco-transformer-base/src/main/resources/application.yaml [hunk 6]
```diff
@@ -48,3 +48,12 @@ management:
...
+    imagemagick:
+      root: ${IMAGEMAGICK_ROOT:/usr/lib64/ImageMagick-7.0.7}
+      dyn: ${IMAGEMAGICK_DYN:/usr/lib64/ImageMagick-7.0.7/lib}
+      exe: ${IMAGEMAGICK_EXE:/usr/bin/convert}
+    pdf_renderer: ${PDF_RENDERER_EXE:/usr/bin/alfresco-pdf-renderer}
+    libreoffice: ${LIBREOFFICE_HOME:/opt/libreoffice6.1}
```
Comment (montgolfiere) [inline alfresco-transformer-base/src/main/resources/application.yaml:58]: Add a sub-property here, in case we need to add other properties in the future.

File: alfresco-transform-imagemagick/alfresco-transform-imagemagick/src/main/java/org/alfresco/transformer/executors/ImageMagickCommandExecutor.java (hunk 2 above)
Comment (KikoUK) [reply alfresco-transform-imagemagick/alfresco-transform-imagemagick/src/main/java/org/alfresco/transformer/executors/ImageMagickCommandExecutor.java:42]: It can't be final in this case as final vars need to be initialised and then won't be able to reassigned it with the new value at runtime.

File: alfresco-transform-pdf-renderer/alfresco-transform-pdf-renderer/src/main/java/org/alfresco/transformer/executors/PdfRendererCommandExecutor.java (hunk 3 above)
Comment (KikoUK) [reply alfresco-transform-pdf-renderer/alfresco-transform-pdf-renderer/src/main/java/org/alfresco/transformer/executors/PdfRendererCommandExecutor.java:38]: It can't be final in this case as final vars need to be initialised and then won't be able to reassigned it with the new value at runtime.

File: alfresco-transform-libreoffice/alfresco-transform-libreoffice/src/main/java/org/alfresco/transformer/executors/LibreOfficeJavaExecutor.java (hunk 4 above)
Comment (KikoUK) [reply alfresco-transform-libreoffice/alfresco-transform-libreoffice/src/main/java/org/alfresco/transformer/executors/LibreOfficeJavaExecutor.java:56]: It can't be final in this case as final vars need to be initialised and then won't be able to reassigned it with the new value at runtime.

File: alfresco-transform-pdf-renderer/alfresco-transform-pdf-renderer/src/main/java/org/alfresco/transformer/executors/PdfRendererCommandExecutor.java (hunk 3 above)
Comment (montgolfiere) [reply alfresco-transform-pdf-renderer/alfresco-transform-pdf-renderer/src/main/java/org/alfresco/transformer/executors/PdfRendererCommandExecutor.java:38]: R u sure - can be initialised once ?

File: alfresco-transform-libreoffice/alfresco-transform-libreoffice/src/main/java/org/alfresco/transformer/executors/LibreOfficeJavaExecutor.java (hunk 4 above)
Comment (montgolfiere) [reply alfresco-transform-libreoffice/alfresco-transform-libreoffice/src/main/java/org/alfresco/transformer/executors/LibreOfficeJavaExecutor.java:56]: Same reply as above ;-)

File: alfresco-transform-imagemagick/alfresco-transform-imagemagick/src/main/java/org/alfresco/transformer/executors/ImageMagickCommandExecutor.java (hunk 2 above)
Comment (montgolfiere) [reply alfresco-transform-imagemagick/alfresco-transform-imagemagick/src/main/java/org/alfresco/transformer/executors/ImageMagickCommandExecutor.java:42]: R u sure - can be initialised once ?

Comment (KikoUK) [reply alfresco-transform-imagemagick/alfresco-transform-imagemagick/src/main/java/org/alfresco/transformer/executors/ImageMagickCommandExecutor.java:42]: It can be initialised to null, but then won't be able to reassigned with the new value at runtime since it's final.

Comment (KikoUK) [reply alfresco-transform-imagemagick/alfresco-transform-imagemagick/src/main/java/org/alfresco/transformer/executors/ImageMagickCommandExecutor.java:42]: Sorry my bad,  just saw "(rather than "static")", if it's no longer static then it can be done.

File: alfresco-transform-core-aio/alfresco-transform-core-aio/src/main/java/org/alfresco/transformer/transformers/ImageMagickAdapter.java [hunk 7]
```diff
@@ -59,10 +60,10 @@
     private static String ID = "imagemagick";
     private ImageMagickCommandExecutor commandExecutor;
 
-    public ImageMagickAdapter() throws Exception 
+    public ImageMagickAdapter(GlobalProperties.ImageMagick execPaths) throws Exception
```
Comment (eknizat) [inline alfresco-transform-core-aio/alfresco-transform-core-aio/src/main/java/org/alfresco/transformer/transformers/ImageMagickAdapter.java:63]: It would be simpler to just pass the 3 strings in, rather than having to worry about another class, especially as this class is external.

File: alfresco-transform-imagemagick/alfresco-transform-imagemagick-boot/src/main/java/org/alfresco/transformer/ImageMagickController.java [hunk 8]
```diff
@@ -79,7 +81,16 @@
 {
     private static final Logger logger = LoggerFactory.getLogger(ImageMagickController.class);
 
-    private ImageMagickCommandExecutor commandExecutor = new ImageMagickCommandExecutor();
+    @Autowired
+    private GlobalProperties externalProps;
```
Comment (eknizat) [inline alfresco-transform-imagemagick/alfresco-transform-imagemagick-boot/src/main/java/org/alfresco/transformer/ImageMagickController.java:85]: Could we just grab the values like @Value("the.value.property : default.value.property")

File: alfresco-transformer-base/src/main/resources/application.yaml [hunk 9]
```diff
@@ -48,3 +48,12 @@ management:
 
 container:
   name: ${HOSTNAME:t-engine}
+
```
Comment (eknizat) [inline alfresco-transformer-base/src/main/resources/application.yaml:51]: You can use the property files in the specific projects: https://github.com/Alfresco/alfresco-transform-core/blob/master/alfresco-transform-libreoffice/alfresco-transform-libreoffice-boot/src/main/resources/application-default.yaml

File: alfresco-transform-core-aio/alfresco-transform-core-aio-boot/src/main/java/org/alfresco/transformer/AIOCustomConfig.java [hunk 10]
```diff
@@ -32,13 +32,31 @@
...
 import org.springframework.context.annotation.Primary;
 
 @Configuration
+@ComponentScan
 public class AIOCustomConfig
 {
+    @Value("${libreoffice.executor.path}")
+    private String libreofficePath;
+
+    @Value("${pdf_renderer.executor.path}")
```
Comment (eknizat) [inline alfresco-transform-core-aio/alfresco-transform-core-aio-boot/src/main/java/org/alfresco/transformer/AIOCustomConfig.java:48]: Very minor, but lets change this to pdfrenderer as it is already like that in some other places

File: alfresco-transform-imagemagick/alfresco-transform-imagemagick/src/main/java/org/alfresco/transformer/executors/ImageMagickCommandExecutor.java [hunk 11]
```diff
@@ -35,9 +35,19 @@
...
 {
-    private static final String ROOT = "/usr/lib64/ImageMagick-7.0.7";
-    private static final String DYN = ROOT + "/lib";
-    private static final String EXE = "/usr/bin/convert";
+    private final String ROOT;
+    private final String DYN;
+    private final String EXE;
+
+    public ImageMagickCommandExecutor(String exe, String dyn, String root)
+    {
```
Comment (eknizat) [inline alfresco-transform-imagemagick/alfresco-transform-imagemagick/src/main/java/org/alfresco/transformer/executors/ImageMagickCommandExecutor.java:43]: Need to add some validation here to check the the values are getting set correctly, and the same goes for all other constructors

File: alfresco-transform-imagemagick/alfresco-transform-imagemagick-boot/src/main/resources/application-default.yaml [hunk 12]
```diff
@@ -2,4 +2,10 @@ queue:
   engineRequestQueue: ${TRANSFORM_ENGINE_REQUEST_QUEUE:org.alfresco.transform.engine.imagemagick.acs}
 transform:
   config:
-    location: classpath:imagemagick_engine_config.json
\ No newline at end of file
+    location: classpath:imagemagick_engine_config.json
+imagemagick:
+  executor:
+    path:
+      root: ${IMAGEMAGICK_ROOT:/usr/lib64/ImageMagick-7.0.7}
```
Comment (eknizat) [inline alfresco-transform-imagemagick/alfresco-transform-imagemagick-boot/src/main/resources/application-default.yaml:9]: So if I want to change the imagemagick.executor.path.root property for example I should have a IMAGEMAGICK_ROOT env property?

Comment (KikoUK) [reply alfresco-transform-imagemagick/alfresco-transform-imagemagick-boot/src/main/resources/application-default.yaml:9]: Yep, you can pass is as a env var like -DIMAGEMAGICK_ROOT=... and this should override the default value

File: alfresco-transform-core-aio/alfresco-transform-core-aio-boot/src/main/java/org/alfresco/transformer/AIOCustomConfig.java [hunk 13]
```diff
@@ -32,13 +32,30 @@
...
 public class AIOCustomConfig
 {
+    @Value("${libreoffice.executor.path}")
+    private String libreofficePath;
+
+    @Value("${pdfrenderer.executor.path}")
+    private String pdfRendererPath;
+
+    @Value("${imagemagick.executor.path.exe}")
+    private String EXE;
```
Comment (eknizat) [inline alfresco-transform-core-aio/alfresco-transform-core-aio-boot/src/main/java/org/alfresco/transformer/AIOCustomConfig.java:51]: Can we change these to camel case variable names which are a little more descriptive

File: alfresco-transform-libreoffice/alfresco-transform-libreoffice-boot/src/main/resources/application-default.yaml [hunk 14]
```diff
@@ -2,4 +2,7 @@ queue:
   engineRequestQueue: ${TRANSFORM_ENGINE_REQUEST_QUEUE:org.alfresco.transform.engine.libreoffice.acs}
 transform:
   config:
-    location: classpath:libreoffice_engine_config.json
\ No newline at end of file
+    location: classpath:libreoffice_engine_config.json
+libreoffice:
```
Comment (eknizat) [inline alfresco-transform-libreoffice/alfresco-transform-libreoffice-boot/src/main/resources/application-default.yaml:6]: We should standardise these properties, this goes for the transform.config.location property as well. Jan suggests transform.core.imagemagick.prop1, So i guess we could have: 
**transform.config.location** > **transform.core.config.location** 
**libreoffice.executor.path** > **transform.core.libreoffice.executor.path**

@montgolfiere @KikoUK  - What do you think?

Comment (KikoUK) [reply alfresco-transform-libreoffice/alfresco-transform-libreoffice-boot/src/main/resources/application-default.yaml:6]: Do you mean to have them under the same props tree transform.core..? I suggested that yesterday and I think you mentioned to keep them separate for now? I prefer the suggestion above, having them together.

Comment (montgolfiere) [reply alfresco-transform-libreoffice/alfresco-transform-libreoffice-boot/src/main/resources/application-default.yaml]: I prefer a well-known prefix - but not too long, eg. do we actually need "executor", eg.

* transform.core.libreoffice.path

File: alfresco-transform-libreoffice/alfresco-transform-libreoffice/src/main/java/org/alfresco/transformer/executors/LibreOfficeJavaExecutor.java (hunk 4 above)
Comment (KikoUK) [reply alfresco-transform-libreoffice/alfresco-transform-libreoffice/src/main/java/org/alfresco/transformer/executors/LibreOfficeJavaExecutor.java:57]: The method accessing this variable is static

File: alfresco-transform-core-aio/alfresco-transform-core-aio-boot/src/main/resources/application-default.yaml [hunk 15]
```diff
@@ -1,2 +1,16 @@
...
+transform:
+  core:
+    pdfrenderer:
+      executor:
+        path: ${PDF_RENDERER_EXE:/usr/bin/alfresco-pdf-renderer}
+    libreoffice:
+      executor:
+        path: ${LIBREOFFICE_HOME:/opt/libreoffice6.3}
+    imagemagick:
+      executor:
```
Comment (montgolfiere) [inline alfresco-transform-core-aio/alfresco-transform-core-aio-boot/src/main/resources/application-default.yaml:12]: Remove "executor/path" ?

File: alfresco-transform-core-aio/alfresco-transform-core-aio-boot/src/main/resources/application-default.yaml [hunk 16]
```diff
@@ -1,2 +1,16 @@
 queue:
-  engineRequestQueue: ${TRANSFORM_ENGINE_REQUEST_QUEUE:org.alfresco.transform.engine.aio.acs}
\ No newline at end of file
+  engineRequestQueue: ${TRANSFORM_ENGINE_REQUEST_QUEUE:org.alfresco.transform.engine.aio.acs}
+transform:
+  core:
+    pdfrenderer:
+      executor:
```
Comment (montgolfiere) [inline alfresco-transform-core-aio/alfresco-transform-core-aio-boot/src/main/resources/application-default.yaml:6]: Remove "executor" ?

File: alfresco-transform-core-aio/alfresco-transform-core-aio-boot/src/main/resources/application-default.yaml [hunk 17]
```diff
@@ -1,2 +1,16 @@
...
+        path: ${PDF_RENDERER_EXE:/usr/bin/alfresco-pdf-renderer}
+    libreoffice:
+      executor:
```
Comment (montgolfiere) [inline alfresco-transform-core-aio/alfresco-transform-core-aio-boot/src/main/resources/application-default.yaml:9]: Remove "executor" ?

File: alfresco-transform-imagemagick/alfresco-transform-imagemagick-boot/pom.xml [hunk 18]
```diff
@@ -81,6 +81,14 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-surefire-plugin</artifactId>
+                <configuration>
+                    <systemPropertyVariables>
+                        <IMAGEMAGICK_EXE>/usr/bin/convert</IMAGEMAGICK_EXE>
+                        <IMAGEMAGICK_DYN>/usr/lib64/ImageMagick-7.0.7/lib</IMAGEMAGICK_DYN>
+                        <IMAGEMAGICK_ROOT>/usr/lib64/ImageMagick-7.0.7</IMAGEMAGICK_ROOT>
```
Comment (montgolfiere) [inline alfresco-transform-imagemagick/alfresco-transform-imagemagick-boot/pom.xml:88]: Should we configure 7.0.7 as a top-level pom version property to help we catch references (at least poms & Dockerfiles) .... we would still need to remember to also update defaults in ootb YAML files.

File: alfresco-transform-libreoffice/alfresco-transform-libreoffice-boot/pom.xml [hunk 19]
```diff
@@ -91,6 +91,11 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-surefire-plugin</artifactId>
+                <configuration>
+                    <systemPropertyVariables>
+                        <LIBREOFFICE_HOME>/opt/libreoffice6.3</LIBREOFFICE_HOME>
```
Comment (montgolfiere) [inline alfresco-transform-libreoffice/alfresco-transform-libreoffice-boot/pom.xml:96]: Should we configure 7.0.7 as a top-level pom version property to help we catch references (at least poms & Dockerfiles) .... we would still need to remember to also update defaults in ootb YAML files.

File: alfresco-transform-libreoffice/alfresco-transform-libreoffice-boot/src/main/resources/application-default.yaml [hunk 20]
```diff
@@ -1,5 +1,9 @@
...
 transform:
-  config:
-    location: classpath:libreoffice_engine_config.json
\ No newline at end of file
+  core:
+    config:
+      location: classpath:libreoffice_engine_config.json
+    libreoffice:
+      executor:
+        path: ${LIBREOFFICE_HOME:/opt/libreoffice6.3}
```
Comment (montgolfiere) [inline alfresco-transform-libreoffice/alfresco-transform-libreoffice-boot/src/main/resources/application-default.yaml:9]: Why LIBREOFFICE_HOME rather than LIBREOFFICE_PATH (or else other way around, eg. transform.core.libreoffice.home) ?

File: alfresco-transform-libreoffice/alfresco-transform-libreoffice/src/main/java/org/alfresco/transformer/executors/LibreOfficeJavaExecutor.java (hunk 4 above)
Comment (montgolfiere) [reply alfresco-transform-libreoffice/alfresco-transform-libreoffice/src/main/java/org/alfresco/transformer/executors/LibreOfficeJavaExecutor.java:57]: OK, thanks. Trivial - could rename to LIBREOFFICE_HOME for consistency with env var ?

File: alfresco-transform-pdf-renderer/alfresco-transform-pdf-renderer-boot/src/main/resources/application-default.yaml [hunk 21]
```diff
@@ -1,5 +1,9 @@
...
 transform:
-  config:
-    location: classpath:pdfrenderer_engine_config.json
\ No newline at end of file
+  core:
+    config:
+      location: classpath:pdfrenderer_engine_config.json
+    pdfrenderer:
+      executor:
+        path: ${PDF_RENDERER_EXE:/usr/bin/alfresco-pdf-renderer}
```
Comment (montgolfiere) [inline alfresco-transform-pdf-renderer/alfresco-transform-pdf-renderer-boot/src/main/resources/application-default.yaml:9]: For consistency (with ImageMagick) why not: remove "executor" and rename "path" to "exe" ?

```
transform:
  core:
    config:
      location: classpath:pdfrenderer_engine_config.json
    pdfrenderer:
      exe: ${PDFRENDERER_EXE:/usr/bin/alfresco-pdf-renderer}
```

Comment (montgolfiere) [reply alfresco-transform-pdf-renderer/alfresco-transform-pdf-renderer-boot/src/main/resources/application-default.yaml:9]: Note: above I've also removed the extra underscore between PDF & RENDERER which potentially can confusion with Spring Boot mapping from dot notation.

File: alfresco-transform-imagemagick/alfresco-transform-imagemagick-boot/pom.xml (hunk 18 above)
Comment (KikoUK) [reply alfresco-transform-imagemagick/alfresco-transform-imagemagick-boot/pom.xml:88]: These property variables are only used in tests as they are part of the surefire-plugin.

File: alfresco-transform-libreoffice/alfresco-transform-libreoffice-boot/pom.xml (hunk 19 above)
Comment (KikoUK) [reply alfresco-transform-libreoffice/alfresco-transform-libreoffice-boot/pom.xml:96]: These property variables are only used in tests as they are part of the surefire-plugin.

File: alfresco-transform-core-aio/alfresco-transform-core-aio-boot/src/main/resources/application-default.yaml [hunk 22]
```diff
@@ -1,2 +1,12 @@
...
+transform:
+  core:
+    pdfrenderer:
+        exe: ${PDFRENDERER_EXE:/usr/bin/alfresco-pdf-renderer}
+    libreoffice:
+        path: ${LIBREOFFICE_HOME:/opt/libreoffice6.3}
+    imagemagick:
+          root: ${IMAGEMAGICK_ROOT:/usr/lib64/ImageMagick-7.0.7}
+          dyn: ${IMAGEMAGICK_DYN:/usr/lib64/ImageMagick-7.0.7/lib}
+          exe: ${IMAGEMAGICK_EXE:/usr/bin/convert}
```
Comment (montgolfiere) [inline alfresco-transform-core-aio/alfresco-transform-core-aio-boot/src/main/resources/application-default.yaml:12]: Trivial - fix yaml indentation of these three (root, dyn, exe), thanks.

File: alfresco-transform-core-aio/alfresco-transform-core-aio/src/main/java/org/alfresco/transformer/transformers/ImageMagickAdapter.java [hunk 23]
```diff
@@ -1,120 +1,121 @@
-/*
```
Comment (montgolfiere) [inline alfresco-transform-core-aio/alfresco-transform-core-aio/src/main/java/org/alfresco/transformer/transformers/ImageMagickAdapter.java:1]: Fix your git/ide config so that we only see the actual diff (check with @PunchedChimera).

Note: EOL/formatting/whitespace changes should be reserved for separate PRs.

File: alfresco-transform-core-aio/alfresco-transform-core-aio/src/main/java/org/alfresco/transformer/transformers/LibreOfficeAdapter.java [hunk 24]
```diff
@@ -1,56 +1,57 @@
-/*
```
Comment (montgolfiere) [inline alfresco-transform-core-aio/alfresco-transform-core-aio/src/main/java/org/alfresco/transformer/transformers/LibreOfficeAdapter.java:1]: Fix your git/ide config so that we only see the actual diff (check with @PunchedChimera).

Note: EOL/formatting/whitespace changes should be reserved for separate PRs.

File: alfresco-transform-core-aio/alfresco-transform-core-aio/src/main/java/org/alfresco/transformer/transformers/PdfRendererAdapter.java [hunk 25]
```diff
@@ -1,79 +1,80 @@
-/*
```
Comment (montgolfiere) [inline alfresco-transform-core-aio/alfresco-transform-core-aio/src/main/java/org/alfresco/transformer/transformers/PdfRendererAdapter.java:1]: Fix your git/ide config so that we only see the actual diff (check with @PunchedChimera).

Note: EOL/formatting/whitespace changes should be reserved for separate PRs.

File: alfresco-transform-core-aio/alfresco-transform-core-aio-boot/src/main/java/org/alfresco/transformer/AIOCustomConfig.java [hunk 26]
```diff
@@ -32,13 +32,30 @@
 import org.alfresco.transformer.transformers.MiscAdapter;
 import org.alfresco.transformer.transformers.PdfRendererAdapter;
 import org.alfresco.transformer.transformers.TikaAdapter;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
```
Comment (eknizat) [inline alfresco-transform-core-aio/alfresco-transform-core-aio-boot/src/main/java/org/alfresco/transformer/AIOCustomConfig.java:37]: unused import

File: alfresco-transform-imagemagick/alfresco-transform-imagemagick-boot/pom.xml (hunk 18 above)
Comment (montgolfiere) [reply alfresco-transform-imagemagick/alfresco-transform-imagemagick-boot/pom.xml:88]: OK, that's fine - we need to remember to change in all places when we next upgrade ImageMagick version.

File: alfresco-transform-libreoffice/alfresco-transform-libreoffice-boot/pom.xml (hunk 19 above)
Comment (montgolfiere) [reply alfresco-transform-libreoffice/alfresco-transform-libreoffice-boot/pom.xml:96]: OK, that's fine - we need to remember to change in all places when we next upgrade LibreOffice version.

File: alfresco-transform-pdf-renderer/alfresco-transform-pdf-renderer-boot/src/main/resources/application-default.yaml [hunk 27]
```diff
@@ -1,5 +1,8 @@
...
   engineRequestQueue: ${TRANSFORM_ENGINE_REQUEST_QUEUE:org.alfresco.transform.engine.alfresco-pdf-renderer.acs}
 transform:
-  config:
-    location: classpath:pdfrenderer_engine_config.json
\ No newline at end of file
+  core:
+    config:
+      location: classpath:pdfrenderer_engine_config.json
+    pdfrenderer:
+        exe: ${PDFRENDERER_EXE:/usr/bin/alfresco-pdf-renderer}
```
Comment (montgolfiere) [inline alfresco-transform-pdf-renderer/alfresco-transform-pdf-renderer-boot/src/main/resources/application-default.yaml:8]: Trivial - fix yaml indentation of these three (root, dyn, exe), thanks.

File: alfresco-transform-libreoffice/alfresco-transform-libreoffice-boot/src/main/resources/application-default.yaml [hunk 28]
```diff
@@ -1,5 +1,8 @@
...
   engineRequestQueue: ${TRANSFORM_ENGINE_REQUEST_QUEUE:org.alfresco.transform.engine.libreoffice.acs}
 transform:
-  config:
-    location: classpath:libreoffice_engine_config.json
\ No newline at end of file
+  core:
+    config:
+      location: classpath:libreoffice_engine_config.json
+    libreoffice:
+        home: ${LIBREOFFICE_HOME:/opt/libreoffice6.3}
```
Comment (montgolfiere) [inline alfresco-transform-libreoffice/alfresco-transform-libreoffice-boot/src/main/resources/application-default.yaml:8]: Trivial - fix yaml indentation of these three (root, dyn, exe), thanks.

File: alfresco-transform-libreoffice/alfresco-transform-libreoffice-boot/src/main/resources/application-default.yaml [hunk 29]
```diff
@@ -1,5 +1,8 @@
 queue:
   engineRequestQueue: ${TRANSFORM_ENGINE_REQUEST_QUEUE:org.alfresco.transform.engine.libreoffice.acs}
 transform:
-  config:
-    location: classpath:libreoffice_engine_config.json
\ No newline at end of file
+  core:
+    config:
+      location: classpath:libreoffice_engine_config.json
+    libreoffice:
```
Comment (eknizat) [inline alfresco-transform-libreoffice/alfresco-transform-libreoffice-boot/src/main/resources/application-default.yaml:7]: It is not enough to just change the transform.core.config.location property in the properties file, This way you have just unset it. You need to update it in the TransformRegistryImpl, otherwise it will just return the empty default config. Keep in mind that all other transformers have the same property too, that needs updating.

Comment (eknizat) [reply alfresco-transform-libreoffice/alfresco-transform-libreoffice-boot/src/main/resources/application-default.yaml:7]: We need to make sure the transform configs are the same before and after these changes.

Comment (KikoUK) [reply alfresco-transform-libreoffice/alfresco-transform-libreoffice-boot/src/main/resources/application-default.yaml:7]: Good catch. I was meant to that.
